src/Symfony/Component/ExpressionLanguage/ParserCache/PSR6ParserCacheAdapter.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\ParserCache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheItemInterface;
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * @author Alexandre GESLIN <alexandre@gesl.in>
+ */
+class PSR6ParserCacheAdapter implements CacheItemPoolInterface
```

Comment:
I'd name this `ParserCacheAdapter`, tag it `@internal` &  `@deprecated` and make it extend from `AbstractAdapter` to reduce the boilerplate of the implementation.